cirq/circuits/circuit_test.py

```diff
@@ -581,44 +579,126 @@ def test_concatenate_with_device():
     assert len(cone) == 0
 
 
+class FakeXmonDevice(cirq.Device):
+    """A Fake XMon device that was created to ensure certain Device validation features are
```

Comment:
I think it's important to note (either here or elsewhere in this file) that what we're testing is *not* the specific behavior of Foxtail and Bristlecone, but rather a set of common behaviors for which these processors are reasonable examples.

Reply:
Yes, I would call this something more specific, like CZOnlyTestDevice or something like that.

Reply:
Right, okay, I will remove the reference to anything Foxtail / Bristlecone and rename the variable names as well. I had originally "ValidatingTestDevice" but that was too generic, maybe I'll go back to that.